R/tmb-methods.R

```diff
@@ -192,34 +194,163 @@ h_get_prediction_variance <- function(object, n_sim, tmb_data) {
 #' model.frame(object)
 #' model.frame(object, include = "subject_var")
 model.frame.mmrm_tmb <- function(formula, data, include = NULL, full, na.action = "na.omit", ...) { # nolint
-  include_choice <- c("subject_var", "visit_var", "group_var", "response_var")
+  # construct updated formula and data arguments
+  lst_formula_and_data <-
+    h_construct_model_frame_inputs(
+      formula = formula,
+      data = data,
+      include = include,
+      full = full
+    )
+
+  # construct data frame to return
+  ret <-
+    stats::model.frame(
+      formula = lst_formula_and_data$formula,
+      data = lst_formula_and_data$data,
+      na.action = na.action
+    )
+  # we need the full formula obs, so recalculating if not already full
+  ret_full <-
+    switch(as.character(lst_formula_and_data$is_full),
+      "TRUE" = ret,
+      "FALSE" =
```

Comment:
Given that `is_full` is a logical flag, should be easier with standard `if else` syntax?

Reply:
updated to use `if {} else {}`